src/cargo/core/compiler/timings.js

```diff
@@ -111,7 +111,7 @@ function render_pipeline_graph() {
     ctx.textAlign = 'start';
     ctx.textBaseline = 'middle';
     ctx.font = '14px sans-serif';
-    const label = `${unit.name}${unit.target} ${unit.duration}s`;
+    const label = `${unit.name} v${unit.version}${unit.target} ${unit.duration}s`;
```

Comment:
> > ![image](https://github.com/rust-lang/cargo/assets/17818024/23aae84e-5107-4352-8d0f-ecdfcac59187)

For me, my biggest concern is how easily can I digest the timing numbers.  I feel that rendering like `quote v1.0.28 0.45s` makes this difficult

Ideas
- @weihanglo mentioned only showing versions on duplicates
- In #7384, parenthesis were used, adding noise but also making it easier to scan
- Tooltips might be a possibility

Any other ideas?

Something else I want to keep in mind is that some improvement is better than no improvement, so if we can't converge on a solution quickly enough, we probably should just move forward.

Reply:
I have no strong opinion, much like @weihanglo. I opened the PR with the simple one line solution exactly for the same reason: displaying the version always is better than not showing at all.

I can implement "showing versions on duplicates" or change the format relatively easily, but tooltips on rendered graphs might be beyond my JS abilities.

Reply:
I can definitely understanding of limited JS abilities; I'm in a similar boat :).  As I said, some improvement is better than none so I didn't want to put too much of a burden on this PR.

Just checked in with @weihanglo and if you are ok with limiting this to dupes, let's go with that.  We can iterate on this further later.

Reply:
If you are open to make versions show only on duplicates, I personally more lean towards that 😆.

Reply:
> I do believe the browser can't search in the graph itself.

Oops. My bad. I forgot it is an SVG.

Reply:
Thanks for trying it out and sharing the results!

I am curious why `syn build script (run)` doesn't have a version because that should be a dupe.

What do you think of adding separator between `labelName` and `duration` so we get a little bit more visual distinction?  Maybe `:` or `-`?

> But I'm still getting some duplicate behavior, apparently we are compiling the same crate twice? 

I suspect this is because we compile for "host" for build scripts and "target" for everything else.  We try to de-duplicate these build steps.  Unsure what is going on that this de-duplication isn't working.  I wonder if its related to the debug info shrinking or something else.

Reply:
> I am curious why syn build script (run) doesn't have a version because that should be a dupe.
Because the target is different. I explicitly used name + target for uniqueness, but that's easy enough to change.

> What do you think of adding separator between labelName and duration so we get a little bit more visual distinction? Maybe : or -?
I like `:`.

@epage @weihanglo
I'm willing to implement any chosen format as long as we decide which one. I personally still lean towards always showing the versions as I find it less confusing (someone looking at this might ask: "why is it showing the version for X and Y but not Z?") but that's just me.

If consistent formatting with the rest of the output is not necessary here, may I suggest one of these formats:
1. `serde_json v1.2.3 syn build script (run): 0.21s`
2. `serde_json (v1.2.3) syn build script (run): 0.21s`
3. `serde_json v1.2.3 (syn build script (run)): 0.21s`
4. `serde_json v1.2.3 syn build script (run) - 0.21s`
5. `serde_json (v1.2.3) syn build script (run) - 0.21s`
6. `serde_json (v1.2.3) (syn build script (run)) - 0.21s`

Bear in mind target can be empty string and, since we showing versions only on duplicate entries, so can it.

Reply:
>  Unsure what is going on that this de-duplication isn't working. I wonder if its related to the debug info shrinking or something else.

Could be. I think we can move on and deal with it afterward.

Regarding the format, I'll leave the choice to you :)

BTW, could you help updates screenshots in that pages? Some of them could be in a better resolution.